### PR TITLE
Feat/#1474 validating behaviour

### PR DESCRIPTION
**Pull request by ciaranschutte**
**Description of changes**

Core component `Button` changes
- changes loading behaviour from css `visibility` toggle to `display` toggle (so the button width will fix its content)
- adds `Loader` prop for custom loading component (previously a spinner was hard coded into button)
- splits styling and typing into their own files

Will add UI comparison once we get this to dev.
Changes Button component so potentially affects a lot of things

**Type of Change**

- [ ] Bug
- [ ] Styling
- [x] New Feature

**Checklist before requesting review:**

- Design (select one):
  - [x] Matches design:
    - component sizes, spacing, and styles
    - font size, weight, colour
    - spelling has been double checked
  - [ ] No design provided, screenshot of changes included in PR
  - [ ] No changes to UI
- UI Kit (select one):
  - [ ] New Components with storybook stories created
  - [x] Modified Existing components and updates stories
  - [ ] No new UIKit components or changes
- [ ] Feature is minimally responsive
- [x] Manual testing of UI feature
- [x] Add copyrights to new files
- [x] Connected ticket to PR

### CODE REVIEW

File: uikit/FocusWrapper/index.tsx [hunk 1]
```diff
@@ -34,10 +32,4 @@ const FocusWrapper = styled('button')`
   }
 `;
 
-FocusWrapper.propTypes = {
```
Comment (ciaranschutte) [inline uikit/FocusWrapper/index.tsx:37]: we have TS now, no need for prop types

File: components/pages/submission-system/common.tsx [hunk 2]
```diff
@@ -47,6 +47,18 @@ export const instructionBoxButtonContentStyle = css`
 export const instructionBoxButtonStyle = css`
   margin-top: 10px;
 `;
+export const instructionBoxLoadingButtonStyle = (theme) => css`
```
Comment (ciaranschutte) [inline components/pages/submission-system/common.tsx:50]: @anncatton This creating of css fragments and passing through irritates me a little. Do you think this is idiomatic usage with Emotion?

File: components/pages/submission-system/program-clinical-submission/Instruction/index.tsx [hunk 3]
```diff
@@ -19,20 +19,49 @@
...
 } from '../../common';
 import FileSelectButton from 'uikit/FileSelectButton';
 import FileTemplatesDownloadButton from './FileTemplatesDownloadButton';
 import Link from 'uikit/Link';
 import { getConfig } from 'global/config';
 import { DOCS_DICTIONARY_PAGE } from 'global/constants/docSitePaths';
 import { useClinicalSubmissionSchemaVersion } from 'global/hooks/useClinicalSubmissionSchemaVersion';
+import { css } from 'uikit';
+import { BUTTON_SIZES } from 'uikit/Button/constants';
+import { ButtonLoader } from 'uikit/Button/types';
```
Comment (samrichca) [inline components/pages/submission-system/program-clinical-submission/Instruction/index.tsx:39]: yay different files for different types of information!

**Review by samrichca — APPROVED** [review]
looks good! very thorough, thanks for splitting up files and getting rid of old crud

**Pull request merged**